algorithms/kernel/svm/oneapi/svm_train_cache_oneapi.h

```diff
@@ -0,0 +1,246 @@
+    const size_t _blockSize;                       /*!< Number of cache lines */
+    const kernel_function::KernelIfacePtr _kernel; /*!< Kernel function */
+};
+
+template <SVMCacheOneAPIType cacheType, typename algorithmFPType>
+class SVMCacheOneAPI
+{};
+
+template <typename algorithmFPType>
+using SVMCacheOneAPIPtr = services::SharedPtr<SVMCacheOneAPIIface<algorithmFPType> >;
+
+/**
+ * No cache: kernel function values are not cached
+ */
+template <typename algorithmFPType>
+class SVMCacheOneAPI<noCache, algorithmFPType> : public SVMCacheOneAPIIface<algorithmFPType>
+{
+    using Helper   = utils::internal::HelperSVM<algorithmFPType>;
+    using super    = SVMCacheOneAPIIface<algorithmFPType>;
+    using thisType = SVMCacheOneAPI<noCache, algorithmFPType>;
+    using super::_kernel;
+    using super::_lineSize;
+    using super::_blockSize;
+
+public:
+    ~SVMCacheOneAPI() {}
+
+    DAAL_NEW_DELETE();
+
+    static SVMCacheOneAPIPtr<algorithmFPType> create(const size_t cacheSize, const size_t blockSize, const size_t lineSize,
+                                                     const NumericTablePtr & xTable, const kernel_function::KernelIfacePtr & kernel,
+                                                     services::Status & status)
+    {
+        status.clear();
+        services::SharedPtr<thisType> res = services::SharedPtr<thisType>(new thisType(blockSize, lineSize, xTable, kernel));
+        if (!res)
+        {
+            status.add(ErrorMemoryAllocationFailed);
+        }
+        else
+        {
+            status = res->init(cacheSize, xTable);
+            if (!status)
+            {
+                res.reset();
+            }
+        }
+        return SVMCacheOneAPIPtr<algorithmFPType>(res);
+    }
+
+    const services::Buffer<algorithmFPType> & getRowsBlock() const override { return _cacheBuff; }
+
+    services::Status compute(const NumericTablePtr & xTable, const services::Buffer<uint32_t> & wsIndices, const size_t p) override
+    {
+        services::Status status;
+        BlockDescriptor<algorithmFPType> xBlock;
+
+        DAAL_CHECK_STATUS(status, xTable->getBlockOfRows(0, xTable->getNumberOfRows(), ReadWriteMode::readOnly, xBlock));
+        const services::Buffer<algorithmFPType> & xBuff = xBlock.getBuffer();
+
+        size_t blockSize                         = _blockSize;
+        services::Buffer<uint32_t> wsIndicesReal = wsIndices;
+        if (_ifComputeSubKernel)
+        {
+            // SubBuffer in not working on L0 (Date: 20200423).
+            // So workaround is set offset _nSelectRows in kernel copyDataByIndices
+            // Uncomment this code after fix
+
+            // blockSize = _blockSize / 2;
+            // wsIndicesReal = wsIndices.getSubBuffer(_nSelectRows, blockSize, &status);
+            // DAAL_CHECK_STATUS_VAR(status);
+            // DAAL_CHECK_STATUS(status, initSubKernel(blockSize, xTable));
```

Comment:
?

Reply:
It need to uncomment, after fix compiler bug.

Reply:
Else we have of all failed tests and incorrect examples.